opentasks/src/main/java/org/dmfs/tasks/utils/AbstractStringCharSequence.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.tasks.utils;
+
+/**
+ * Base class for {@link CharSequence}s that use a {@link String} delegate value.
+ *
+ * @author Gabor Keszthelyi
+ */
+// TODO Remove this class when the Factory, Lazy, AbstractCachingCharSequence line is available in java tools library
+public abstract class AbstractStringCharSequence implements CharSequence
+{
+    private final String mValue;
+
+
+    public AbstractStringCharSequence(String value)
```

Comment:
Why does it take a String? Couldn't it just take any other `CharSequence`?

Reply:
Also, can this be merged with `AbstractCharSequence`

Reply:
These were in part leftover from earlier and I didn't want to spend time on them since we would have support for this from jems, but I've reworked them now.
There is `DelegatingCharSequence`, and `CachingCharSequence` which uses `OnDemand` and `Cached`. I chose `OnDemand` here, but let me know if you already have another name for that.